src/helpers/query-daily-data.tsx

```diff
@@ -0,0 +1,94 @@
+﻿import { add } from "date-fns";
+import getDayKey from "./get-day-key";
+import { DailyDataType, DailyDataTypeDefinition } from "./daily-data-types";
+import allTypeDefinitions from "./daily-data-types/all";
+import { FontAwesomeSvgIcon } from "react-fontawesome-svg-icon";
+import { faPersonRunning } from "@fortawesome/free-solid-svg-icons";
+import React from "react";
+import { defaultFormatter } from "./daily-data-types/formatters";
+
+export type DailyDataQueryResult = { [key: string]: number };
+export type DailyDataProvider = (startDate: Date, endDate: Date) => Promise<DailyDataQueryResult>;
+export type DailyDataAvailabilityCheck = (modifiedAfter?: Date) => Promise<boolean>;
+
+let dailyDataTypes = new Map(
+	allTypeDefinitions.map((typeDefinition) => [typeDefinition.type, typeDefinition])
+);;
+
+export function registerDailyDataProvider(type: string, provider: DailyDataProvider, availabilityCheck: DailyDataAvailabilityCheck) {
+	dailyDataTypes.set(type, {
+		type: type,
+		getLabel: () => type,
+		dataProvider: provider,
+		availabilityCheck: availabilityCheck,
+		icon: <FontAwesomeSvgIcon icon={faPersonRunning} />,
```

Comment:
Do we want this to be hard coded to the same icon for all data providers?

Reply:
Ah I added a comment above here; so this method should essentially be deprecated in favor of "registerDailyDataTypeDefinition" which lets you specify all those things by sending in a DailyDataTypeDefinition itself.

I just kept this method for backwards compatibility; I could add additional arguments but I think the answer is more to just use the new method if you need to specify this stuff.

Ultimately, if you're currently using the registerDailyDataProvider method and upgrade MyDataHelpsUI, this stuff shouldn't really affect you unless you use one of the new components anyways

Reply:
Sounds good.